### ISSUE TITLE
inconsistent representation of commitment ATX and node ID

### ISSUE THREAD

**Issue by poszu** [spacemeshos/post#163]
## Problem statement
Node ID and commitment ATX ID are formatted in base64 both in postdata_metadata.json and in post-rs CLI initializer. However, postcli accepts them in hex. They are also formatted in hex when logged in go-spacemesh. It is inconsistent and causes extra work to convert.

## DoD
We should format ATXes in the same way everywhere  (either hex or base64 - this is to decide).

**Comment by lrettig** [comment]
The node ID is stored in hex in key.bin and base64 in postdata_metadata.json. This is really confusing and frustrating.

**Comment by fasmat** [comment]
> The node ID is stored in hex in key.bin and base64 in postdata_metadata.json. This is really confusing and frustrating.

key.bin contains the private key hex-encoded. `postdata_metadata.json` contains the NodeID (public key) in base64.

**Comment by lrettig** [comment]
Right. IMHO they should both be in hex :) (The private key includes the public key so at least this way you can do pattern-matching.)

**Comment by fasmat** [comment]
I also responded in this comment: https://github.com/spacemeshos/post/issues/203#issuecomment-1655330443

Updating this is more involved than it initially appears. We need a migration plan such that existing PoSTs are updated from the current schema to the new one. This is btw. true for any file layout we want to change and is probably worth documenting a general approach for as a cross cutting concern 🙂

**Comment by fasmat** [comment]
We already had that issue come up in that some PoSTs have a `NonceValue` in their metadata and some not: https://github.com/spacemeshos/post/pull/202

Thankfully there was an easy migration path that allowed to just add the value if not already present.
